src/archivematicaCommon/lib/storageService.py

```diff
@@ -238,6 +238,36 @@ def get_files_from_backlog(files):
     return copy_files(backlog, processing, files)
 
 
+# ############ SPACES #############
+
+def get_space(access_protocol=None, path=None):
+    """ Returns a list of storage spaces, optionally filtered by parameters.
+
+    Queries the storage service and returns a list of storage spaces,
+    optionally filtered by access_protocol or path.
+
+    access_protocol: How the storage is accessed.  Should reference storage
+        service purposes, in storage_service.locations.models.py
+    """
+    return_spaces = []
+    url = _storage_service_url() + 'space/'
+    params = {
+        'access_protocol': access_protocol,
+        'path': path,
+        'offset': 0,
+    }
+    while True:
+        response = _storage_api_session().get(url, params=params)
+        spaces = response.json()
+        return_spaces += spaces['objects']
+        if not spaces['meta']['next']:
```

Comment:
We're trusting in a response field to end this `while` loop, which someday may end up being an endless loop. Do we have a way to get the total count of spaces from the SS?

Reply:
I take your point, but we're trusting that the tastypie API to always return a boolean for `['meta']['next']` and to eventually return `False`. I think this is ok. The weaker argument is that this is how this was done previously (this is old code) and how it still is done in other parts of this module. It's moot now though because I'm removing this code in favour of that which indicates encryption status using ES AIP indexing.